.github/workflows/test.yml

```diff
@@ -150,13 +149,33 @@ jobs:
           sed -i "s#^test.arango.js.*#test.arango.js=$ARANGO_JS#" test.cfg
           cat test.cfg
 
-      - name: Run tests
-        shell: bash
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: 7.6
+
+      - name: build the war file (requires compileJava, javadoc)
+        run: gradle war --info
+
+      - name: run the ${{matrix.test_cmd}} tests and generate coverage
         run: |
-          ant javadoc
-          ant ${{matrix.ant_test}}
+          mkdir test-reports
+          if [ "${{matrix.test_cmd}}" = "test_quick_coverage" ]; then
+            ant "${{matrix.test_cmd}}"
+          else
+            gradle "coverage${{matrix.test_cmd}}" --info
+          fi
 
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v3
+        if: ${{ matrix.test_cmd }} == 'test_quick_coverage'
         with:
           fail_ci_if_error: true
+
+      - name: upload jacoco report
+        uses: actions/upload-artifact@v3
+        if: ${{ matrix.test_cmd }} != 'test_quick_coverage'
+        with:
+          name: coverage${{matrix.test_cmd}}-${{matrix.mongo}}.exec
+          path: build/
+          if-no-files-found: warn
```

Comment:
temporarily add a step to make the `build` directory available for download so that the test and coverage reports can be seen